grails-app/services/org/broadinstitute/orsp/IssueService.groovy

```diff
@@ -191,6 +192,22 @@ class IssueService {
         }
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
+    Issue modifyExtraProperties(Object input, String projectKey) {
+        Issue issue = queryService.findByKey(projectKey)
+        Collection<IssueExtraProperty> newProperties = getSingleValuedPropsForSaving(issue, input)
+        List<IssueExtraProperty> extraPropertiesList = new ArrayList<>()
+        newProperties.collect {
+            property ->
+                extraPropertiesList.add(property)
+        }
```

Comment:
This block is a little confusing. `newProperties.collect` returns a collection of whatever happens in the closure. Since you're just calling `add`, I think `newProperties` is just a collection of booleans. I think you want to:

Replace:
```
        Collection<IssueExtraProperty> newProperties = getSingleValuedPropsForSaving(issue, input)
        List<IssueExtraProperty> extraPropertiesList = new ArrayList<>()
        newProperties.collect {
            property ->
                extraPropertiesList.add(property)
        }
```
With:
```
        Collection<IssueExtraProperty> extraPropertiesList = getSingleValuedPropsForSaving(issue, input)
```

Reply:
This is very helpful, I didn't notice that! this is changed now